Slim/Web/Settings/Server/Behavior.pm

```diff
@@ -89,50 +86,80 @@ sub handler {
 		$prefs->set('releaseTypesToIgnore', [ keys %releaseTypesToIgnore ]);
 
 		foreach my $role (Slim::Schema::Contributor::defaultContributorRoles()) {
-			$prefs->set(lc($role)."AlbumLink", $paramRef->{"pref_".lc($role)."AlbumLink"});
+			$prefs->set(lc($role)."AlbumLink", $paramRef->{"pref_".lc($role)."AlbumLink"} ? "1" : "0");
 			next if $role eq "ALBUMARTIST" || $role eq "ARTIST";
-			$prefs->set(lc($role)."InArtists", $paramRef->{"pref_".lc($role)."InArtists"});
+			$prefs->set(lc($role)."InArtists", $paramRef->{"pref_".lc($role)."InArtists"} ? "1" : "0");
 		}
 		foreach my $role (Slim::Schema::Contributor::userDefinedRoles()) {
 			$userDefinedRoles->{$role}->{albumLink} = $paramRef->{"pref_".lc($role)."AlbumLink"};
 			$userDefinedRoles->{$role}->{include} = $paramRef->{"pref_".lc($role)."InArtists"};
 		}
 		$prefs->set('userDefinedRoles', $userDefinedRoles);
-		$userDefinedRoles = $prefs->get('userDefinedRoles');
+
+		# custom role handling
+		my $currentRoles = $prefs->get('userDefinedRoles');
+		my $customRoleId = Slim::Schema::Contributor->getMinCustomRoleId();
+
+		my $customTags = {};
+		my $changed = 0;
+
+		foreach my $pref (keys %{$paramRef}) {
+			if ($pref =~ /(.*)_tag$/) {
+				my $key = $1;
+				my $tag = uc($paramRef->{$pref});
+
+				if ( $tag ) {
+					$customTags->{$tag} = {
+						name => $paramRef->{$key . '_name'} || $tag,
+						id => $currentRoles->{$tag} ? $currentRoles->{$tag}->{id} : $customRoleId++,
+						include => $currentRoles->{$tag}  ? $currentRoles->{$tag}->{include} : 1,
+						albumLink => $currentRoles->{$tag}  ? $currentRoles->{$tag}->{albumLink} : 1,
+					};
+
+					if ( !$currentRoles->{$tag} || $currentRoles->{$tag}->{name} ne $customTags->{$tag}->{name} ) {
+						Slim::Utils::Strings::storeExtraStrings([{
+							strings => { EN => $customTags->{$tag}->{name}},
+							token   => $tag,
+						}]);
+						$changed = 1;
+					}
+				}
+			}
+		}
+
```

Comment:
Could this be consolidate with the plugin's handler? Eg. make it a method accessible to both?

Reply:
I was planning on pulling the code out of the plugin, no need for the maintenance functionality in both places.